mteb/abstasks/languages.py

```diff
@@ -14,3 +17,61 @@
 
 with path_to_lang_scripts.open("r") as f:
     ISO_TO_SCRIPT = json.load(f)
+
+
+@dataclass
+class LangScriptFilter:
+    language_scripts: set[str]
+    scripts: set[str]
+    languages: set[str]
+
+    @classmethod
+    def from_languages_and_scripts(
+        cls, languages: list[str] | None, scripts: list[str] | None
+    ) -> LangScriptFilter:
+        lang_script_codes = set()
+        # normalize to 3 letter language codes
+        normalized_langs = set()
+        filter_lang = languages is not None
+
+        if filter_lang:
+            for lang in languages:
+                lang_script = lang.split("-")
+
+                is_lang_script_code = len(lang_script) == 2
+                if is_lang_script_code:
+                    normalized_langs.add(lang_script[0])
+                    lang_script_codes.add(lang)
+                else:
+                    normalized_langs.add(lang)
+
+        filter_scripts = scripts is not None
+        script_codes: set[str] = set(scripts) if filter_scripts else set()
+        return cls(
+            language_scripts=lang_script_codes,
+            scripts=script_codes,
+            languages=normalized_langs,
+        )
+
+    def is_valid_language(self, language: str) -> bool:
```

Comment:
Does this (and `is_valid_script`) function check for containment instead of validity? Would something like `contains_language` be more fitting?

Reply:
Have renamed langscriptfilter to LanguageScripts and used contains (I think it is better as well)

```
LanguageScripts.contains_language(...) 
```